src/main.js

```diff
@@ -385,24 +412,63 @@ class WebSearchNavigator {
     const getOpt = (key) => {
       return this.options.sync.get(key);
     };
-    this.register(getOpt('nextKey'), () => {
-      if (!getOpt('autoSelectFirst') && this.isFirstNavigation) {
-        this.resultsManager.focus(0);
-        this.isFirstNavigation = false;
-      } else {
-        this.resultsManager.focusNext(getOpt('wrapNavigation'));
-      }
-      return false;
-    });
-    this.register(getOpt('previousKey'), () => {
-      if (!getOpt('autoSelectFirst') && this.isFirstNavigation) {
-        this.resultsManager.focus(0);
-        this.isFirstNavigation = false;
-      } else {
-        this.resultsManager.focusPrevious(getOpt('wrapNavigation'));
-      }
-      return false;
-    });
+    if (!this.searchEngine.gridNavigation) {
+      this.register(getOpt('nextKey'), () => {
+        if (!getOpt('autoSelectFirst') && this.isFirstNavigation) {
+          this.resultsManager.focus(0);
+          this.isFirstNavigation = false;
+        } else {
+          this.resultsManager.focusNext(getOpt('wrapNavigation'));
+        }
+        return false;
+      });
+      this.register(getOpt('previousKey'), () => {
+        if (!getOpt('autoSelectFirst') && this.isFirstNavigation) {
+          this.resultsManager.focus(0);
+          this.isFirstNavigation = false;
+        } else {
+          this.resultsManager.focusPrevious(getOpt('wrapNavigation'));
+        }
+        return false;
```

Comment:
The first two `this.register` call apply to both grid and non-grid navigation, right?
If so, they should only be called once for both (and the code inside the if should only bind `navigatePreviousResultPage`  and `navigateNextResultPage`).

Reply:
Yes the register calls are for the same keys in both grid and non-grid navigation but the functions being called are different.
This is because currently the up and down arrows (although correct for navigating through a vertical list of search results) are used for horizontal navigation on the youtube home page, which just feels unnatural. Hence the two repeating calls for the same keys in different cases

Reply:
Yes, my bad- looking at the diff I missed these differences. Since these 6 snippets of code are some similar
(previously it was 2), maybe it makes sense to create a function?
Something like:

```javascript
const onFocusChange = (callback) => {
  return () => {
    if (!getOpt('autoSelectFirst') && this.isFirstNavigation) {
      this.resultsManager.focus(0);
      this.isFirstNavigation = false;
    } else {
      callback(getOpt('wrapNavigation'));
    }
    return false;
  };
};
this.register(
    getOpt('nextKey'),
    onFocusChange(this.resultsManager.focusNext),
);
```